pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -291,323 +435,764 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 
-	// for ContentTree config Publisher
-	pubContentTreeConfig, err := ps.NewPublication(
-		pubsub.PublicationOptions{
-			AgentName: agentName,
-			TopicType: types.ContentTreeConfig{},
-		})
+	getconfigCtx.pubPhysicalIOAdapters, err = ps.NewPublication(pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: types.PhysicalIOAdapterList{},
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	pubContentTreeConfig.ClearRestarted()
-	getconfigCtx.pubContentTreeConfig = pubContentTreeConfig
+	getconfigCtx.pubPhysicalIOAdapters.ClearRestarted()
 
-	// for volume config Publisher
-	pubVolumeConfig, err := ps.NewPublication(
-		pubsub.PublicationOptions{
-			AgentName: agentName,
-			TopicType: types.VolumeConfig{},
-		})
+	getconfigCtx.pubDevicePortConfig, err = ps.NewPublication(pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: types.DevicePortConfig{},
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	pubVolumeConfig.ClearRestarted()
-	getconfigCtx.pubVolumeConfig = pubVolumeConfig
 
-	// for disk config Publisher
-	pubDisksConfig, err := ps.NewPublication(
-		pubsub.PublicationOptions{
-			AgentName: agentName,
-			TopicType: types.EdgeNodeDisks{},
-		})
+	// Publish NetworkXObjectConfig and for outselves. XXX remove
+	getconfigCtx.pubNetworkXObjectConfig, err = ps.NewPublication(pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: types.NetworkXObjectConfig{},
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	pubDisksConfig.ClearRestarted()
-	getconfigCtx.pubDisksConfig = pubDisksConfig
 
-	// for Edge Node Info Publisher
-	pubEdgeNodeInfo, err := ps.NewPublication(
-		pubsub.PublicationOptions{
-			AgentName:  agentName,
-			TopicType:  types.EdgeNodeInfo{},
-			Persistent: true,
-		})
+	getconfigCtx.pubNetworkInstanceConfig, err = ps.NewPublication(pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: types.NetworkInstanceConfig{},
+	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	pubEdgeNodeInfo.ClearRestarted()
-	getconfigCtx.pubEdgeNodeInfo = pubEdgeNodeInfo
 
-	// Look for global config such as log levels
-	subGlobalConfig, err := ps.NewSubscription(pubsub.SubscriptionOptions{
-		AgentName:     agentName,
-		MyAgentName:   agentName,
-		TopicImpl:     types.ConfigItemValueMap{},
-		Persistent:    true,
-		Activate:      false,
-		Ctx:           &zedagentCtx,
-		CreateHandler: handleGlobalConfigCreate,
-		ModifyHandler: handleGlobalConfigModify,
-		DeleteHandler: handleGlobalConfigDelete,
-		WarningTime:   warningTime,
-		ErrorTime:     errorTime,
+	getconfigCtx.pubAppInstanceConfig, err = ps.NewPublication(pubsub.PublicationOptions{
+		AgentName: agentName,
+		TopicType: types.AppInstanceConfig{},
 	})
 	if err != nil {
 		log.Fatal(err)
 	}
-	zedagentCtx.subGlobalConfig = subGlobalConfig
-	subGlobalConfig.Activate()
+	getconfigCtx.pubAppInstanceConfig.ClearRestarted()
+	// XXX defer this until we have some config from cloud or saved copy
+	getconfigCtx.pubAppInstanceConfig.SignalRestarted()
```

Comment:
Seems you changed the order here.

Reply:
Correct, this should go after device is onboarded. Fixed